python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -0,0 +1,112 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("TrailedSourceFilterTask", "TrailedSourceFilterConfig")
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.utils.timer import timeMethod
+
+
+class TrailedSourceFilterConfig(pexConfig.Config):
+    """Config class for TrailedSourceFilterTask.
+    """
+
+    maxTrailLength = pexConfig.Field(
+        dtype=float,
+        doc="Length of long trailed sources to remove from the input catalog, "
+            "in arcseconds per second. Default comes from DMTN-199, which "
+            "requires removal of sources with trails longer than 10 "
+            "degrees/day, which is 36000/3600/24arcsec/second, or roughly"
+            "0.416 arcseconds per second.",
+        default=36000/3600.0/24.0,
+    )
+
+
+class TrailedSourceFilterTask(pipeBase.Task):
+    """Find trailed sources in DIASources and filter them as per DMTN-199
+    guidelines.
+
+    This task checks the length of trailLength in the DIASource catalog using
+    a given arcsecond/second rate from maxTrailLength and the exposure time.
+    The two values are used to calculate the maximum allowed trail length and
+    filters out any trail longer than the maximum. The maxTrailLength is
+    outlined in DMTN-199 and determines the default value.
+    """
+
+    ConfigClass = TrailedSourceFilterConfig
+    _DefaultName = "trailedSourceFilter"
+
+    @timeMethod
+    def run(self, dia_sources, exposure_time):
+        """Remove trailed sources longer than ``config.maxTrailLength`` from
+        the input catalog.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            New DIASources to be checked for trailed sources.
+        exposure_time : `float`
+            Exposure time from difference image.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components.
+
+            - ``dia_sources`` : DIASource table that is free from unwanted
+             trailed sources. (`pandas.DataFrame`)
+
+            - ``trailed_dia_sources`` : DIASources that have trails which
+            exceed maxTrailLength/second*exposure_time.
+            (`pandas.DataFrame`)
+        """
+
+        trail_mask = self._check_dia_source_trail(dia_sources, exposure_time)
+
+        return pipeBase.Struct(
+            diaSources=dia_sources[~trail_mask].reset_index(drop=True),
+            trailedDiaSources=dia_sources[trail_mask].reset_index(drop=True))
+
+    def _check_dia_source_trail(self, dia_sources, exposure_time):
+        """Find DiaSources that have long trails.
+
+        Creates a mask for sources with lengths greater than 0.416
+        arcseconds/second multiplied by the exposure time.
```

Comment:
Again, don't mention the default value, since it's configurable.

```suggestion
        Return a mask of sources with lengths greater than ``config.maxTrailLength``  multiplied by the exposure time.
```

Reply:
Swapped the wording to what you've suggested.